### PR TITLE
Add a bit more control over persistence behavior.

### PR DESCRIPTION
**Pull request by jessemyers**
Allows:
 - Merge behavior to be customized.
 - Database name to be customized.

### CODE REVIEW

File: microcosm_postgres/store.py [hunk 1]
```diff
@@ -167,6 +164,12 @@ def search(self, *criterion, **kwargs):
         query = self._filter(query, **kwargs)
         return query.all()
 
+    def expunge(self, instance):
+        return self.session.expunge(instance)
+
+    def merge(self, instance, new_instance):
```
Comment (netama) [inline microcosm_postgres/store.py]: why is this getting `instance`? conventions?

File: microcosm_postgres/factories.py [hunk 2]
```diff
@@ -29,7 +30,10 @@ def configure_sqlalchemy_engine(graph):
 
     """
     # use different database name for testing
-    if graph.metadata.testing:
+    if graph.config.postgres.database_name is not None:
```
Comment (netama) [inline microcosm_postgres/factories.py]: nice :)

**Comment by netama** [comment]
LGTM, not sure what one of the changes is for

File: microcosm_postgres/store.py (hunk 1 above)
Comment (jessemyers) [inline microcosm_postgres/store.py]: I need these for some other work I'm doing with polymorphic types. It turns out that if you `PATCH` a polymorphic object but don't specify the column it's polymorphic *on*, things fail.

Comment (netama) [inline microcosm_postgres/store.py]: sounds fun

Comment (jessemyers) [inline microcosm_postgres/store.py]: In the case I care about, we need to manually populate the new instance with an (immutable) value from the old instance that it doesn't have.

**Pull request closed without merging**